.github/workflows/ci.yaml

```diff
@@ -28,47 +28,44 @@ jobs:
       - name: Run pre-commit checks
         uses: pre-commit/action@v3.0.0
 
-  linux:
-    name: "Linux - unit tests - Python ${{ matrix.PYTHON_VERSION }}"
+  unit-tests:
+    name: "unit tests - Python ${{ matrix.PYTHON_VERSION }} - SA ${{ matrix.SA_VERSION }}"
     runs-on: ubuntu-latest
     env:
       CI: True
     strategy:
       fail-fast: true
       matrix:
-        PYTHON_VERSION: ['3.8', '3.9', '3.10']
+        # PYTHON_VERSION: ['3.8', '3.9', '3.10']
+        PYTHON_VERSION: ['3.10']
+        SA_VERSION: ['<=2.0', '>=2.0']
     steps:
       - name: Checkout branch
         uses: actions/checkout@v3
         with:
           ref: ${{ github.head_ref }}
-      - uses: conda-incubator/setup-miniconda@v2
+      - name: Setup conda
+        uses: mamba-org/provision-with-micromamba@v15
         with:
-          python-version: ${{ matrix.PYTHON_VERSION }}
-          miniforge-variant: Mambaforge
-          miniforge-version: 4.11.0-2
-          use-mamba: true
-          environment-file: environment.yml
-          activate-environment: datajudge
-      - name: Run Unit Tests
-        shell: bash -l {0}
-        run: |
-          flit install -s
-          pytest --cov=datajudge --cov-report=xml --cov-append tests/unit
-      - name: Generate code coverage report
-        uses: codecov/codecov-action@v3.1.1
+          extra-specs: |
+            python=${{ matrix.PYTHON_VERSION }}
+            sqlalchemy${{ matrix.SA_VERSION }}
+      - name: Run tests
+        uses: ./.github/actions/pytest
         with:
-          file: ./coverage.xml
+          args: "tests/unit"
 
-  linux-integration_tests-sqlserver:
-    name: "Linux - integration tests - Python ${{ matrix.PYTHON_VERSION }} - mssql"
+  sqlserver:
+    if: ${{ contains(github.event.pull_request.labels.*.name, 'sqlserver') || contains(github.event.pull_request.labels.*.name, 'ready') }}
+    name: "Python ${{ matrix.PYTHON_VERSION }} - SA ${{ matrix.SA_VERSION }} - mssql"
```

Comment:
This filter checks if the PR has a label `sqlserver` (for example) or if the PR has a label `ready` and runs the following test accordingly.
Actually saves a lot of time, more than I thought.